### PR TITLE
Feat adui 4842 gather examples in same files #2

### PR DESCRIPTION
**Pull request by dmgauthier**
### Proposed Changes

gathered Calendar with date pickers and headers with breadcrumbheaders

### Potential Breaking Changes

only affect the demo

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/react-vapor/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

**Comment by gdostie** [comment]
I think some `// start-print` and `// stop-print` directives would cleanup the source code displayed in the demo quite a bit.

File: packages/react-vapor/src/components/headers/examples/HeaderExamples.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+export class HeaderExamples extends React.Component<any, any> {
+    render() {
+        return (
+            <Section>
+                <SimpleHeader />
+                <BreadcrumbHeaders />
+            </Section>
+        );
+    }
+}
```
Comment (GermainBergeron) [inline packages/react-vapor/src/components/headers/examples/HeaderExamples.tsx]:
```typescript
export const HeaderExamples: React.FunctionComponent = () => (
    <Section>
        <SimpleHeader />
        <BreadcrumbHeaders />
    </Section>
);
```

File: packages/react-vapor/src/components/headers/examples/HeaderExamples.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            <Section level={3} title="Basic header with actions and tabs">
+                <BasicHeader
+                    title={defaultTitle}
+                    description="Simple description for the title"
+                    actions={actions}
+                    tabs={defaultTabs}
+                />
+            </Section>
+        </Section>
+        <Section level={2}></Section>
```
Comment (GermainBergeron) [inline packages/react-vapor/src/components/headers/examples/HeaderExamples.tsx]: Is this section usefull?
```suggestion
```

File: packages/react-vapor/src/components/calendar/examples/CalendarDatePickerDateSelectionExamples.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    render() {
+        return (
+            <Section>
+                <CalendarComponent />
+                <DatePickerComponents />
+                <DatesSelectionComponent />
+            </Section>
+        );
+    }
+}
```
Comment (GermainBergeron) [inline packages/react-vapor/src/components/calendar/examples/CalendarDatePickerDateSelectionExamples.tsx]:
```typescript
export const CalendarDatePickerDatesSelectionExamples: React.FunctionComponent = () => (
    <Section>
        <CalendarComponent />
        <DatePickerComponents />
        <DatesSelectionComponent />
    </Section>
);
CalendarDatePickerDatesSelectionExamples.title = 'Calendar, DatePicker and DatesSelection';
```

File: packages/react-vapor/src/components/calendar/examples/CalendarDatePickerDateSelectionExamples.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
```
Comment (GermainBergeron) [inline packages/react-vapor/src/components/calendar/examples/CalendarDatePickerDateSelectionExamples.tsx]:
```suggestion
// start-print
```

File: packages/react-vapor/src/components/headers/examples/HeaderExamples.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    render() {
+        return (
+            <Section>
+                <SimpleHeader />
+                <BreadcrumbHeaders />
+            </Section>
+        );
+    }
+}
+
```
Comment (GermainBergeron) [inline packages/react-vapor/src/components/headers/examples/HeaderExamples.tsx]:
```suggestion
// start-print
```

File: packages/react-vapor/src/components/headers/examples/HeaderExamples.tsx (hunk 2 above)
Comment (dmgauthier) [reply packages/react-vapor/src/components/headers/examples/HeaderExamples.tsx]: VERY USEFULL :see_no_evil:

File: packages/react-vapor/src/components/calendar/examples/CalendarDatePickerDateSelectionExamples.tsx (hunk 3 above)
Comment (dmgauthier) [reply packages/react-vapor/src/components/calendar/examples/CalendarDatePickerDateSelectionExamples.tsx]: :+1:

File: packages/react-vapor/src/components/calendar/examples/CalendarDatePickerDateSelectionExamples.tsx (hunk 4 above)
Comment (dmgauthier) [reply packages/react-vapor/src/components/calendar/examples/CalendarDatePickerDateSelectionExamples.tsx]: :+1:

File: packages/react-vapor/src/components/headers/examples/HeaderExamples.tsx (hunk 5 above)
Comment (dmgauthier) [reply packages/react-vapor/src/components/headers/examples/HeaderExamples.tsx]: :+1:

**Comment by Charles-ErickBG** [comment]
@dmgauthier 
What do you think about removing the last 2 examples (without padding and without border-bottom)?

I think we might not want to front these props that much and they speak for themselves in case we really need to use them (hasPadding={false}, hasBorderBottom={false}).

**Comment by Charles-ErickBG** [comment]
I would change the title of the Datepicker section for:
Screen Shot 2019-10-28 at 1 25 05 PM


Datepicker

Calendar and other related components are meant to pick dates, so it should be intuitive enough this way IMO.

**Comment by Charles-ErickBG** [comment]
I think we should remove the Breadcrumbs section now that it is combined in the Header section.

Screen Shot 2019-10-28 at 1 27 27 PM
Screen Shot 2019-10-28 at 1 27 39 PM

**Comment by Charles-ErickBG** [comment]
Screen Shot 2019-10-28 at 1 29 21 PM

Since the other pages/components don't have a title, I would remove this one for now.

**Comment by dmgauthier** [comment]
> @dmgauthier
> What do you think about removing the last 2 examples (without padding and without border-bottom)?
> 
> I think we might not want to front these props that much and they speak for themselves in case we really need to use them (hasPadding={false}, hasBorderBottom={false}).

I the demo was not open source, I would say yes for the `hasPadding={false}`. And since some times the border bottom of the header will overlap with the borderTop of the table, I strongly suggest that we keep the `hasBorderBottom={false}`

**Pull request merged**